Modules/Core/Sources/Core/Buffer/Buffer.swift

```diff
@@ -6,7 +6,136 @@ import Darwin
 #endif
 
 public typealias Byte = UInt8
-public typealias Buffer = DispatchData
+
+public struct Buffer : RandomAccessCollection {
+    public typealias Iterator = BufferIterator
+    public typealias Index = Int
+    public typealias Indices = DefaultRandomAccessIndices<Buffer>
+    
+    public private(set) var bytes: ContiguousArray<Byte>
+    
+    public var count: Int {
+        return bytes.count
+    }
+    
+    public init(_ bytes: [Byte] = []) {
+        self.bytes = ContiguousArray(bytes)
+    }
+    
+    public init(bytes: [Byte]) {
+        self.bytes = ContiguousArray(bytes)
+    }
+    
+    public init(bytes buffer: UnsafeBufferPointer<UInt8>) {
+        self.bytes = ContiguousArray(buffer)
+    }
+    
+    public mutating func append(_ other: Buffer) {
+        bytes += other.bytes
```

Comment:
Maybe `append` would be a tiny bit faster? I think `+=` calls `append` under the covers.